source/src/molecular-unfolding/cdk/utils/utils-images.ts

```diff
@@ -31,49 +37,100 @@ export enum ECRRepoNameEnum {
 
 
 export class ECRImageUtil {
-
-  public static newInstance() {
-    return new this();
+  public static newInstance(scope: Construct) {
+    return new this(scope);
   }
 
-  private constructor() {
+  private scope: Construct
+
+  private constructor(scope: Construct) {
+    this.scope = scope;
   }
 
   public getECRImage(name: ECRRepoNameEnum): ecs.ContainerImage | lambda.DockerImageCode {
 
+    const usePreBuildImages = process.env.SOLUTION_PRE_BUILD_IMAGES || false;
+    const version = process.env.SOLUTION_VERSION || 'v1.0.0';
+    const imagePrefix = process.env.IMAGE_PREFIX || '';
+    const ecrAccount = process.env.SOLUTION_ECR_ACCOUNT || '';
+    const repoName = process.env.SOLUTION_ECR_REPO_NAME || '';
+    const region = Stack.of(this.scope).region;
+
     if (name == ECRRepoNameEnum.Batch_Create_Model) {
-      return ecs.ContainerImage.fromAsset(
-        path.join(__dirname, '../../'), {
-          file: 'batch-images/create-model/Dockerfile',
-        });
+
+      if (usePreBuildImages) {
+        return ecs.ContainerImage.fromEcrRepository(ecr.Repository.fromRepositoryAttributes(this.scope, 'Batch_Create_Model', {
```

Comment:
prefer to extract method to simplify the duplicated code

Reply:
changed